File: backend/models/file.py
from pony import orm
from pony.orm import *
from .models import db
from datetime import datetime


class File(db.Entity):
    title = Required(LongStr)
    title_short = Optional(str)
    title_word_count = Optional(int)
    number = Required(str)
    publish_date = Required(datetime)
    type = Required(str)
    author = Required(str)
    file_size = Optional(int, size=64)
    word_count = Optional(int)
    insert_date = Required(datetime)
    link = Required(str)
    pages = Optional(int)

File: api/cmd/parlismonitoring-api/models/file.go
package models

/*
 * Tag... - a very simple struct
 */
type File struct {
    Id              NInt        `json:"id"`
    Title           NString     `json:"title"`
    Number          NString     `json:"number"`
    TitleWordCount  NInt        `json:"title_word_count"`
    PublishDate     NTime       `json:"publish_date"`
    Type            NString     `json:"type"`
    Author          NString     `json:"author"`
    FileSize        NInt        `json:"file_size"`
    WordCount       NInt        `json:"word_count"`
    InsertDate      NTime       `json:"insert_date"`
    Link            NString     `json:"link"`
    Pages           NInt        `json:"pages"`
}